lib/protobuf/field/field_hash.rb

```diff
@@ -43,9 +43,13 @@ def merge!(other)
       alias update merge!
 
       # Return a hash-representation of the given values for this field type.
-      # The value in this case would be the hash itself.
+      # The value in this case would be the hash itself, right? Unfortunately
+      # not because the values of the map could be messages themselves that we
+      # need to transform.
```

Comment:
it seems to be the convention for `to_hash` elsewhere, for example for messages:

```
    # Return a hash-representation of the given fields for this message type.
    def to_hash
      result = {}

      @values.each_key do |field_name|
        value = self[field_name]
        field = self.class.get_field(field_name, true)
        hashed_value = value.respond_to?(:to_hash_value) ? value.to_hash_value : value
        result[field.name] = hashed_value
      end

      result
    end
```